UI/src/view/linear_asset/laneModellingForm.js

```diff
@@ -0,0 +1,1540 @@
+
+      this.addField = function(field, sideCode){
+        if(!formFields['sidecode_'+sideCode])
+          formFields['sidecode_'+sideCode] = [];
+        formFields['sidecode_'+sideCode].push(field);
+      };
+
+      this.getFields = function(sideCode){
+        if(!formFields['sidecode_'+sideCode])
+          Error("The form of the sidecode " + sideCode + " doesn't exist");
+        return formFields['sidecode_'+sideCode];
+      };
+
+      this.getAllFields = function(){
+        return _.flatten(_.map(formFields, function(form) {
+          return form;
+        }));
+      };
+
+      this.removeFields = function(sideCode){
+        if(!formFields['sidecode_'+sideCode])
+          throw Error("The form of the sidecode " + sideCode + " doesn't exist");
+
+        formFields['sidecode_'+sideCode] = [];
+      };
+    };
+
+    var forms = new AvailableForms();
+
+    me.initialize = function(assetTypeConfiguration, feedbackModel){
+      var rootElement = $('#feature-attributes');
+      _assetTypeConfiguration = assetTypeConfiguration;
+      lanesAssets = _assetTypeConfiguration.selectedLinearAsset;
+
+      eventbus.on('laneModellingForm: reload', function() {
+        reloadForm($('#feature-attributes'));
+      });
+
+      function setInitialForm() {
+        defaultFormStructure = formStructure;
+        currentFormStructure = mainLaneFormStructure;
+        isAddByRoadAddressActive = false;
+        lanesAssets.setCurrentLane(parseInt(_.min(_.map(_assetTypeConfiguration.selectedLinearAsset.get(), function (lane) {
+          return _.head(_.find(lane.properties, function (property) {
+            return property.publicId == "lane_code";
+          }).values).value;
+        }))));
+      }
+
+      new FeedbackDataTool(feedbackModel, assetTypeConfiguration.layerName, assetTypeConfiguration.authorizationPolicy, assetTypeConfiguration.singleElementEventCategory);
+
+      var updateStatusForMassButton = function(element) {
+        if(assetTypeConfiguration.selectedLinearAsset.isSplit()) {
+          element.prop('disabled', !(me.isSaveable() && me.isSplitOrSeparatedAllowed()));
+        } else
+          element.prop('disabled', !(me.isSaveable()));
+      };
+
+      eventbus.on(events('selected'), function () {
+        var isDisabled = false;
+        setInitialForm();
+        rootElement.find('#feature-attributes-header').html(me.renderHeader(_assetTypeConfiguration.selectedLinearAsset));
+        rootElement.find('#feature-attributes-form').html(me.renderForm(_assetTypeConfiguration.selectedLinearAsset, isDisabled));
+        rootElement.find('#feature-attributes-form').prepend(me.renderPreview(_assetTypeConfiguration.selectedLinearAsset));
+        rootElement.find('#feature-attributes-form').append(me.renderLaneButtons(_assetTypeConfiguration.selectedLinearAsset));
+        rootElement.find('#feature-attributes-footer').html(me.renderFooter(_assetTypeConfiguration.selectedLinearAsset));
+      });
```

Comment:
why not call reloadForm function here instead of rootElement.find(..) ?

Reply:
Done